### PR TITLE
NIAD-148: create scaffold for HA/GP link adaptor service inbouns

### PR DESCRIPTION
**Pull request by Alicja-Kainos**
related to https://github.com/nhsconnect/integration-adaptor-common/pull/21

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -42,21 +40,48 @@ def start_tornado_server() -> None:
...
+        urls=['amqp://localhost:5672'],
+        queue='nhais_inbound',
+        username='quest',
+        password='quest',
+        max_retries='3',
+        retry_delay=0.1,
+        get_message_callback=message_callback)
+
+async def send_message_to_queue(adaptor):
+    #temporary code, to be replaced by proper tests
```
Comment (Alicja-Kainos) [inline main.py]: the sending does not work now, this method is only for test/debug purposes and should be removed

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

File: inbound/request/inbound_handler.py [hunk 2]
```diff
@@ -17,11 +21,25 @@ def __init__(self):
...
-    async def on_message_recieved(self, edifact_message):
-        if edifact_message.message:
+    def message_callback(self, edifact_message):
+        logger.info(f'Message recieved by inbound, message: {edifact_message}')
+        if edifact_message:
             fhir_message = self.edifact_to_fhir.convert_edifact_to_fhir(edifact_message)
             self.inbound_sequence_number_manager.record_sequence_number(fhir_message)
             self.edifact_recep_producer.generate_sequences(fhir_message)
-            self.supplier_incoming_mq.send(fhir_message)
+            loop = asyncio.new_event_loop()
```
Comment (mattd-kainos) [inline inbound/request/inbound_handler.py]: I'm concerned about this code creating and setting the event loop. set_event_loop is applied globally and will affect other parts of the application

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
concerns over manual management of event loop

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
just a basic integration test to demonstrate the inbound queues working correctly

File: integration_tests/int_inbound_tests.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            max_retries=int(config.get_config('INBOUND_QUEUE_MAX_RETRIES', default='3')),
+            retry_delay=int(config.get_config('INBOUND_QUEUE_RETRY_DELAY', default='100')) / 1000)
+
+    @test_utilities.async_test
+    async def test_acceptance_transaction(self):
+        mymessage = 'this is my message'
+        await self.incoming_queue.send_async(mymessage)
+        message = self.supplier_queue.get_next_message_on_queue()
+        self.assertIsNotNone(message, 'message from queue should exist')
+        self.assertTrue(len(message.body) > 0, 'message from queue should not be empty')
```
Comment (mattd-kainos) [inline integration_tests/int_inbound_tests.py:45]: what should the supplier_queue message body be exactly?

File: README.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ when the service is run.
 | NHAIS_OUTBOUND_QUEUE_PASSWORD    |          | (Optional) password for the amqp server for outbound (to mesh) message queue
 | NHAIS_OUTBOUND_QUEUE_MAX_RETRIES | 3        | The number of times a request to the outbound (to mesh) broker(s) will be retried
 | NHAIS_OUTBOUND_QUEUE_RETRY_DELAY | 100      | Milliseconds delay between retries to the outbound (to mesh) broker(s)
-| NHAIS_DYNAMODB_ENDPOINT_URL      |          | URL of dynamodb instance (if used)
+| NHAIS_DB_ENDPOINT_URL      |          | URL of dynamodb instance (if used)
```
Comment (michael-kainos) [inline README.md:26]: as the variable does not now reference dynamodb should we update the description also to remove this?

**Review by mattd-kainos — APPROVED** [review]

**Review by Broom94 — APPROVED** [review]

**Pull request merged**